init-tests-after-clone.sh

```diff
@@ -1,18 +1,73 @@
-#!/usr/bin/env bash
+#!/bin/sh
 
-set -e
+set -eu
 
-if [[ -z "$TRAVIS" ]]; then
-  read -rp "This operation will destroy locally modified files. Continue ? [N/y]: " answer
-  if [[ ! $answer =~ [yY] ]]; then
-    exit 2
-  fi
+fallback_repo_for_tags='https://github.com/gitpython-developers/GitPython.git'
+
+ci() {
+    # For now, check just these, as a false positive could lead to data loss.
+    test -n "${TRAVIS-}" || test -n "${GITHUB_ACTIONS-}"
+}
+
+no_version_tags() {
+    test -z "$(git tag -l '[0-9]*' 'v[0-9]*')"
+}
+
+warn() {
+    if test -n "${GITHUB_ACTIONS-}"; then
+        printf '::warning ::%s\n' "$*" >&2  # Annotate workflow.
```

Comment:
This also has the benefit of creating warning annotations on the workflow run itself, [as can be seen here](https://github.com/EliahKagan/GitPython/actions/runs/6400756535) (below the error annotations which are from the `mypy` step that always fails but has `continue-on-error` set).

I had meant to include a link to that in my PR description, but forgot... which actually points at one limitation of annotations on a workflow itself: because CI results shown in PRs and on commits link to specific job results, the page for a workflow run that shows workflow-run annotations often goes unexamined.

Reply:
Recently, GitHub also shows [annotations in the GitHub Actions log view](https://github.blog/changelog/2024-10-01-annotations-in-the-github-actions-log-view/), so annotations on a workflow are more discoverable now.